lib/events/events.go

```diff
@@ -314,19 +309,26 @@ func (s *bufferedSubscription) pollingLoop() {
 		s.buf[s.next] = ev
 		s.next = (s.next + 1) % len(s.buf)
 		s.cur = ev.SubscriptionID
-		s.cond.Broadcast()
+		if s.newEventNotification != nil {
+			close(s.newEventNotification)
+			s.newEventNotification = nil
+		}
 		s.mut.Unlock()
 	}
 }
 
-func (s *bufferedSubscription) Since(id int, into []Event) []Event {
+func (s *bufferedSubscription) Since(id int, into []Event, timeout time.Duration) []Event {
+	eventAvailable := s.waitForNewEventWithTimeout(id, timeout)
+	if !eventAvailable {
```

Comment:
It could, but since it returns a "magic" bool with no obvious meaning I wanted to give it a name, for readability. Happy to remove it if you care strongly.